cilium-cli/clustermesh/clustermesh.go

```diff
@@ -1222,8 +1223,11 @@ func (k *K8sClusterMesh) processSingleRemoteClient(ctx context.Context, remoteCl
 	}
 
 	// Expand those values to include the clustermesh configuration
-	newCluster := getCluster(aiRemote, !match)
-	state.localNewClusters = append(state.localNewClusters, newCluster)
+	newClusterName, newCluster := getCluster(aiRemote, !match)
+	if state.localNewClusters == nil {
+		state.localNewClusters = make(map[string]any)
+	}
+	state.localNewClusters[newClusterName] = newCluster
```

Comment:
Should we warn if cluster with this name already existed?

Reply:
I made it hard fail since we have some similar error here: https://github.com/cilium/cilium/blob/main/cilium-cli/clustermesh/clustermesh.go#L540 It probably should fail here too IMO